awx/playbooks/project_update.yml

```diff
@@ -111,6 +114,11 @@
             url_password: "{{ scm_password|default(omit) }}"
             force_basic_auth: true
           register: get_archive
+          retries: 12
+          delay: 10
+          until: >
+            get_archive is succeeded or
+            get_archive.status_code is defined 
```

Comment:
This would increase the time it takes for any project update to fail due to unrelated problems, and this would make the user experience worse. You've done some investigating into what the timeout conditions look like, and I would like to see the `until` condition reflect those specifics. Like, could it retry until the status_code is not one of (504, 408, maybe others?). That would still allow for quick fails in the event that something is misconfigured.

Reply:
You said

> Any other error like [403] Authentication issues, or [404] File not found or [5xx] errors will make the task fail immediately.

But I'm not following how. If status_code is 403, why wouldn't this retry?

Reply:
But let me run and share some test results...